### PR TITLE
APEX-204 #resolve #comment setting linebreaks to true for NoWhitespac…

### PR DESCRIPTION
**Pull request by otterc**
…eBefore->dot and continuation indentation to be 2

### CODE REVIEW

**Comment by ilganeli** [comment]
@chandnisingh 
I've identified another inconsistency:
Braces placement for static statements throws an error saying that it should be on next line. However, try, if, while, etc. are all presumed to be same line. This is also inconsistent with method declarations (which are assumed to be next-line by style and code guide).

**Comment by otterc** [comment]
@ilganeli  that is not an inconsistency. We adopted K&R style and the placement of braces is according to that.
https://en.wikipedia.org/wiki/Indent_style#K.26R_style

**Comment by ilganeli** [comment]
@chandnisingh Understood, either way, code style should be updated to reflect this.

**Comment by otterc** [comment]
Sorry which code style are you referring to?  :-)

**Comment by ilganeli** [comment]
Sorry, to be clear, in IntelliJ the apex-style format sets the following code:

```
static 
{
...
...
}
```

As

```
static {
...
...
}
```

The first is what's actually valid according to the CheckStyle.

**Comment by otterc** [comment]
got it

**Comment by vrozov** [comment]
@ilganeli please update APEX-204 JIRA to add discrepancy for the static block.
@chandnisingh can you please check what is supported for static block in all major IDEs

**Comment by otterc** [comment]
@ilganeli @vrozov  I have created another ticket for that issue
https://malhar.atlassian.net/browse/APEX-211

This pull request addresses the issues currently in APEX-204 and can be merged

**Comment by vrozov** [comment]
@chandnisingh I'd prefer to address all known checkstyle issues in a single commit.

**Comment by otterc** [comment]
@vrozov the other issue is not a checkstyle issue. It is a bug with checked in templates.

**Comment by vrozov** [comment]
@chandnisingh please check whether all IDEs support separate settings for static blocks, otherwise it may be necessary to adjust checkstyle.

**Comment by otterc** [comment]
@vrozov placement of brace is according to K&R style and all IDEs support that.

**Comment by vrozov** [comment]
I don't see option in IntelliJ to change brace location for static or synchronized. Both fall into other settings. I think we need to change code style and have static to be consistent with synchronized()

```
static {
...
}

synchronized() {
...
}
```

**Comment by otterc** [comment]
@vrozov static blocks are defined within a class scope and from what I read about K&R style anything within a class scope should left brace placement in the next line.

Anything within a method scope should have left brace at the eol.

**Comment by vrozov** [comment]
The code style is not in stone. It needs to be convenient for developers to follow and adhere. Both static and synchronized() designate block of code and it seems that IntelliJ formats both the same way and there is no separate setting for static and/or synchronized().

I can't find reference to K&R style regarding class/method scope. Here is reference to [google code style](https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style). Note that braces follow the Kernighan and Ritchie style ("Egyptian brackets") for nonempty blocks and **block-like constructs**:
- No line break before the opening brace.
- Line break after the opening brace.
- Line break before the closing brace.
- Line break after the closing brace if that brace terminates a statement or the body of a method, constructor or named class. For example, there is no line break after the brace if it is followed by else or a comma.

**Comment by ilganeli** [comment]
Vlad - I am concerned that if all check styles are a single commit it will be overwhelmingly large. I think one commit per small batch of files is more reasonable.

Thank you,
Ilya Ganelin

-----Original Message-----

**Comment by tweise** [comment]
I think we can do multiple sweeps by area and a final one that catches everything left.

**Comment by vrozov** [comment]
@ilganeli Ilya, sorry for not being clear, I mean that I prefer to fix checkstyle definition issues (apex_check.xml) in one single commit (and particular static/synchronized block style), it does not apply to .java files that have code style violations. There is a separate e-mail thread and JIRA filed to address existing large number of code style violations in .java where majority of community seems to prefer #1 option that means one single commit. I voted for #3.

**Comment by otterc** [comment]
I checked Eclipse and Intellij and they don't differentiate between blocks within a class and method so I am going to push the change.

**Comment by tweise** [comment]
@vrozov Vlad, my interpretation of support of option 1 is the desire to fix all issues in one go instead of sometime in the future, but not necessarily a single commit.

**Comment by vrozov** [comment]
@tweise Thomas, can you please bring this to e-mail thread on dev@apex.incubator.apache.org.

File: engine/pom.xml [hunk 1]
```diff
@@ -145,7 +145,7 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-checkstyle-plugin</artifactId>
         <configuration>
-          <maxAllowedViolations>2320</maxAllowedViolations>
+          <maxAllowedViolations>2063</maxAllowedViolations>
```
Comment (vrozov) [inline engine/pom.xml:148]: Is the number reduced due to change to the apex_checks.xml or code style violations were fixed as part of other commits?

**Comment by otterc** [comment]
@ilganeli Can you please merge this pull request. Thanks

File: apex_checks.xml [hunk 2]
```diff
@@ -54,13 +54,13 @@
     <module name="LeftCurly">
       <property name="option" value="eol"/>
       <property name="tokens"
-                value="LITERAL_CATCH, LITERAL_DO, LITERAL_ELSE, LITERAL_FINALLY, LITERAL_FOR, LITERAL_IF, LITERAL_SWITCH, LITERAL_SYNCHRONIZED, LITERAL_TRY, LITERAL_WHILE"/>
+                value="LITERAL_CATCH, LITERAL_DO, LITERAL_ELSE, LITERAL_FINALLY, LITERAL_FOR, LITERAL_IF, LITERAL_SWITCH, LITERAL_SYNCHRONIZED, LITERAL_TRY, LITERAL_WHILE, STATIC_INIT"/>
```
Comment (vrozov) [inline apex_checks.xml:57]: Is there a way to enforce brace on a new line for anonymous class definitions like in

```
    Thread t = new Thread(new Runnable() 
    { // brace on the new line is not enforced
      @Override
      public void run()
      {

      }
    });
```

**Comment by vrozov** [comment]
@chandnisingh please squash commits before merge

**Comment by otterc** [comment]
why? the 2 commits address 2 distinct issues then why should we squash them. Actually if you check other apache projects, it is recommended not to squash the commits when the change was a result of a review or another issue which was brought up.

**Comment by vrozov** [comment]
This was an existing practice. If we are going to change it, let's put it for the vote.

**Comment by tweise** [comment]
I see only one JIRA referenced? But in general, when you submit the good old patch to some other projects, that's a single commit. And in the main repository, we want to see the net change for a given issue and not the many different things that I might have tried in my repo to get there.

**Comment by tweise** [comment]
Nevermind, I see you piggyback another ticket. And I thought we are doing one event at a time...

File: engine/pom.xml (hunk 1 above)
Comment (otterc) [inline engine/pom.xml:148]: The violations were fixed as part of other commits which I don't know. Whoever fixed them didn't change this number

**Comment by ilganeli** [comment]
@tweise @chandnisingh This looks good to merge, for future I agree with @tweise that we should try to avoid solving multiple issues with one PR where possible (granted I know that's my fault here).

**Pull request merged**